### ISSUE TITLE
Remove SELF_CONTAINED as a Catalog type

### ISSUE THREAD

**Issue by matthewhanson** [stac-utils/pystac#299]
catalog_type distinguishes between ABSOLUTE_PUBLISHED, RELATIVE_PUBLISHED, and SELF_CONTAINED.

However the only different between RELATIVE_PUBLISHED and SELF_CONTAINED is that there is no `self` link on the root catalog of a SELF_CONTAINED catalog.

I don't think this is enough of a difference to warrant calling out as a catalog type.  For one, the absolute self link at root isn't really used in most (all?) tooling other than for user to know where the root cat lives. I guess stac-browser uses it to display and create the absolute link for it.

You can always just ignore the self link or remove it entirely.  Or we could just have an option to save it or now when you save a catalog (include_self_on_root).  This is clearer than having a catalog type where it isn't obvious what the practical difference is.

**Comment by lossyrob** [comment]
I'd want to see this change in [the spec](https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#self-contained-catalogs) before implementing in PySTAC, so if we were to go this route it would be after 1.0.  Perhaps move this to a stac-spec issue?

**Comment by gadomski** [comment]
I find `SELF_CONTAINED` very useful, e.g. when creating example catalogs for **stactools-packages**: https://github.com/stactools-packages/usda-cdl/blob/688e7f3defe0e2f8dfdb235eb019d07cf2bebe88/scripts/create_examples.py. So I'm 👎🏽 on this change. @matthewhanson this is a pretty old ticket -- do you still agree with your past self?

**Comment by gadomski** [comment]
I agree with your general sentiment about catalog types, but what pystac has aligns with the best practices: https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#self-contained-catalogs. IMO any changes should be made to the best practices document first, then trickle down to pystac?
